### ISSUE TITLE
Accept json data from stdin

### ISSUE THREAD

**Issue by blaggacao** [spiffe/spire#1864]
It would be really handy, if spire-server would accept reading `-data` from `stdin`.

Currently, I must copy any gitops-managed json files into the server container before being able to apply them.

Alternatively, I could parse the json myself and transpose it into cli arguments, but that defies the possibility to specify json objects.

The bash snipped I wised to work from our gitops repo tooling:

```console
$ echo "$(yq r -j {{yaml}})" | kubectl -n {{spireNamespace}} exec {{serverPod}} -- {{spireBin}} entry create -node -data -
```
or more concise:

```console
$ echo "$(yq r -j {{yaml}})" | kubectl -n {{spireNamespace}} exec {{serverPod}} -- {{spireBin}} entry create -node -
```

**Comment by evan2645** [comment]
Hi @blaggacao, thank you for opening this!

Given that the entry create command already supports `-data` flag, simply extending it to understand the special `-` value seems super reasonable and useful.

I would imagine that if/when the `-data` flag is specified, we don't want any other create-related flags to be set... and we should probably validate that this is the case.

In the first example you shared above, I can see that `-node` flag is set. Since this is a flag that normally relates to a specific entry being created, I'd prefer to see it included in the list of mutual exclusivity with the `-data` flag. I believe that today, any additional flags are ignored if `-data` is set... maybe that's good enough for now?

**Comment by blaggacao** [comment]
> Given that the entry create command

_Create_ & _Update_. Correct.

> any additional flags are ignored if -data is set... maybe that's good enough for now?

Yes that's alright. Mutual exclusivity to be checked would be a great educative addition, though.

**Comment by blaggacao** [comment]
_I figured out the `-node` flag in https://github.com/spiffe/spire/pull/1868 after opening this issue._

**Comment by evan2645** [comment]
👍 sounds good. A contribution would be welcome if you were willing.

**Comment by blaggacao** [comment]
I'm in a leading position amidst a project in motion, for the moment (and some while), I can only leave breadcrumb trails. :+1:

**Comment by evan2645** [comment]
No problem @blaggacao thank you for capturing this

**Comment by hiyosi** [comment]
@blaggacao 
@evan2645 

I've tried to implement the feature, Does the PR( #1905 ) solve this issue? 
If so, I'll turn the PR status to ready for review.

**Comment by evan2645** [comment]
This issue has been solved by @hiyosi in #1905 

Thank you very much @hiyosi for the contribution!
